test/unit/lib/plugins/aws/deployFunction.test.js

```diff
@@ -421,6 +425,232 @@ describe('AwsDeployFunction', () => {
         })
       ).to.be.equal(true);
     });
+
+    describe('#KMSKeyArn should be set by "kmsKeyArn" or "awsKmsKeyArn"', () => {
+      const lambdaUpdateConfigStub = sinon.stub();
+      const lambdaUpdateFunctionCodeStub = sinon.stub();
+      const awsRequestStubMap = {
+        Lambda: {
+          getFunction: {
+            Configuration: {
+              CodeSha256: 'agaswetdsadfv345ee',
+            },
+          },
+          updateFunctionConfiguration: lambdaUpdateConfigStub,
+          updateFunctionCode: lambdaUpdateFunctionCodeStub,
+        },
+        IAM: {
+          getRole: {
+            Arn: 'arn:aws:iam::123456789012:role/Admin',
+          },
+        },
+      };
+
+      beforeEach(() => {
+        lambdaUpdateConfigStub.resetHistory();
+        lambdaUpdateFunctionCodeStub.resetHistory();
+      });
+
+      it('configuration uses the "kmsKeyArn" instead of functionObj.awsKmsKeyArn', async () => {
+        await runServerless({
+          fixture: 'function',
+          cliArgs: ['deploy', 'function', '--function', 'foo'],
+          lastLifecycleHookName: 'deploy:function:deploy',
+          awsRequestStubMap,
+          configExt: {
+            functions: {
+              foo: {
+                handler: 'index.handler',
+                name: 'foobar',
+                awsKmsKeyArn: 'arn:aws:kms:us-east-1:oldKey',
+              },
+            },
+            provider: {
+              vpc: {
```

Comment:
Are these extra `vpc`, `environment`, `memorySize` and `timeout` variables needed here? I think if not, let's leave them out as they're not important in this particular test case